From b4dc774c230d82ba7d9d417528f5fe8ea8532e71 Mon Sep 17 00:00:00 2001
From: Valentin Raevsky <valentin@compulab.co.il>
Date: Mon, 1 Jun 2020 21:09:18 +0300
Subject: [PATCH 41/45] mcm-imx8m-mini: Separate the sb and the som stuff

Add sb-mcm-imx8m-mini.dtsi file.
All non SoM stuff move to the sb-mcm-imx8-rev1.dtsi.

Signed-off-by: Valentin Raevsky <valentin@compulab.co.il>
---
 arch/arm64/boot/dts/compulab/Makefile              |   3 +-
 arch/arm64/boot/dts/compulab/mcm-imx8m-mini.dts    | 138 ++-----------------
 arch/arm64/boot/dts/compulab/sb-mcm-imx8-rev1.dtsi | 148 +++++++++++++++++++++
 .../arm64/boot/dts/compulab/sbc-mcm-imx8m-mini.dts |   9 ++
 4 files changed, 171 insertions(+), 127 deletions(-)
 create mode 100644 arch/arm64/boot/dts/compulab/sbc-mcm-imx8m-mini.dts

diff --git a/arch/arm64/boot/dts/compulab/Makefile b/arch/arm64/boot/dts/compulab/Makefile
index 61f6ed05e10d..dfc1daf6da4b 100644
--- a/arch/arm64/boot/dts/compulab/Makefile
+++ b/arch/arm64/boot/dts/compulab/Makefile
@@ -1,10 +1,11 @@
 dtb-$(CONFIG_ARCH_FSL_IMX8MM) += ucm-imx8m-mini.dtb
 dtb-$(CONFIG_ARCH_FSL_IMX8MM) += ucm-imx8m-mini-thermal.dtb
-dtb-$(CONFIG_ARCH_FSL_IMX8MM) += ucm-imx8m-mini-eb-dsi2lvds.dts
+dtb-n += ucm-imx8m-mini-eb-dsi2lvds.dts
 dtb-$(CONFIG_ARCH_FSL_IMX8MM) += ucm-imx8m-mini-dsi2.dtb
 dtb-$(CONFIG_ARCH_FSL_IMX8MM) += ucm-imx8m-mini-m4.dtb
 dtb-$(CONFIG_ARCH_FSL_IMX8MM) += mcm-imx8m-mini.dtb
 dtb-$(CONFIG_ARCH_FSL_IMX8MM) += mcm-imx8m-mini-thermal.dtb
+dtb-$(CONFIG_ARCH_FSL_IMX8MM) += sbc-mcm-imx8m-mini.dtb
 
 always         := $(dtb-y)
 subdir-y       := $(dts-dirs)
diff --git a/arch/arm64/boot/dts/compulab/mcm-imx8m-mini.dts b/arch/arm64/boot/dts/compulab/mcm-imx8m-mini.dts
index 82b4a666727a..556e7c67d5f1 100644
--- a/arch/arm64/boot/dts/compulab/mcm-imx8m-mini.dts
+++ b/arch/arm64/boot/dts/compulab/mcm-imx8m-mini.dts
@@ -15,7 +15,6 @@
 /dts-v1/;
 
 #include "../freescale/fsl-imx8mm.dtsi"
-#include "sb-mcm-imx8-rev1.dtsi"
 
 / {
 	model = "CompuLab MCM i.MX8MM board";
@@ -117,38 +116,6 @@
 		vin-supply = <&regulator_bt_reg_on>;
 	};
 
-	simple_sound: sound {
-		compatible = "simple-audio-card";
-		simple-audio-card,name = "mcm-imx8m-mini";
-		simple-audio-card,widgets =
-			"Headphone", "Headphone Jack",
-			"Line", "Line Out",
-			"Microphone", "Mic Jack",
-			"Line", "Line In";
-		simple-audio-card,routing =
-			"Headphone Jack", "RHPOUT",
-			"Headphone Jack", "LHPOUT",
-			"MICIN", "Mic Bias",
-			"Mic Bias", "Mic Jack";
-		simple-audio-card,format = "i2s";
-		simple-audio-card,bitclock-master = <&sound_master>;
-		simple-audio-card,frame-master = <&sound_master>;
-		simple-audio-card,bitclock-inversion;
-
-		sound_master: simple-audio-card,cpu {
-			sound-dai = <&sai2>;
-			system-clock-frequency = <0>;
-			system-clock-direction = "out";
-		};
-
-		sound_codec: simple-audio-card,codec {
-			sound-dai = <&wm8731>;
-			system-clock-direction = "in";
-			system-clock-type = "mclk";
-		};
-
-	};
-
 	backlight {
 		compatible = "pwm-backlight";
 		pwms = <&pwm2 0 3000000 0>;
@@ -437,6 +404,17 @@
 				MX8MM_IOMUXC_GPIO1_IO13_PWM2_OUT	0x03
 			>;
 		};
+
+		pinctrl_flexspi0: flexspi0grp {
+			fsl,pins = <
+				MX8MM_IOMUXC_NAND_ALE_QSPI_A_SCLK		0x1c2
+				MX8MM_IOMUXC_NAND_CE0_B_QSPI_A_SS0_B		0x82
+				MX8MM_IOMUXC_NAND_DATA00_QSPI_A_DATA0		0x82
+				MX8MM_IOMUXC_NAND_DATA01_QSPI_A_DATA1		0x82
+				MX8MM_IOMUXC_NAND_DATA02_QSPI_A_DATA2		0x82
+				MX8MM_IOMUXC_NAND_DATA03_QSPI_A_DATA3		0x82
+			>;
+		};
 	};
 };
 
@@ -621,88 +599,7 @@
 	clock-frequency = <400000>;
 	pinctrl-names = "default";
 	pinctrl-0 = <&pinctrl_i2c4>;
-	status = "okay";
-
-	wm8731: wm8731@1a {
-		#sound-dai-cells = <0>;
-		compatible = "wlf,wm8731";
-		reg = <0x1a>;
-		status = "okay";
-	};
-
-	ov5640_mipi: ov5640_mipi@3c {
-		compatible = "ovti,ov5640_mipi";
-		reg = <0x3c>;
-		status = "okay";
-		pinctrl-names = "default";
-		pinctrl-0 = <&pinctrl_csi1>;
-		clocks = <&clk IMX8MM_CLK_CLKO1_DIV>;
-		clock-names = "csi_mclk";
-		assigned-clocks = <&clk IMX8MM_CLK_CLKO1_SRC>,
-				  <&clk IMX8MM_CLK_CLKO1_DIV>;
-		assigned-clock-parents = <&clk IMX8MM_CLK_24M>;
-		assigned-clock-rates = <0>, <24000000>;
-		csi_id = <0>;
-		/*
-		To be looked into & fixed
-		As of now the device is alwas out of reset & under the power
-		*/
-		pwn-gpios = <&pca9555 8 GPIO_ACTIVE_LOW>;
-		rst-gpios = <&pca9555 3 GPIO_ACTIVE_HIGH>;
-
-		mclk = <24000000>;
-		mclk_source = <0>;
-		port {
-			ov5640_mipi1_ep: endpoint {
-				remote-endpoint = <&mipi1_sensor_ep>;
-			};
-		};
-	};
-	goodix_ts@5d {
-		compatible = "goodix,gt911";
-		reg = <0x5d>;
-
-		interrupt-parent = <&gpio1>;
-		interrupts = <1 IRQ_TYPE_LEVEL_HIGH>;
-
-		pinctrl-names = "default";
-		pinctrl-0 = <&touchscreen_pins>;
-
-		irq-gpios = <&gpio1 1 GPIO_ACTIVE_HIGH>;
-		reset-gpios = <&pca9555 4 GPIO_ACTIVE_HIGH>;
-		status = "okay";
-	};
-
-};
-
-&mipi_csi_1 {
-	#address-cells = <1>;
-	#size-cells = <0>;
-	status = "okay";
-	port {
-		mipi1_sensor_ep: endpoint1 {
-			remote-endpoint = <&ov5640_mipi1_ep>;
-			data-lanes = <2>;
-			csis-hs-settle = <13>;
-			csis-clk-settle = <2>;
-			csis-wclk;
-		};
-
-		csi1_mipi_ep: endpoint2 {
-			remote-endpoint = <&csi1_ep>;
-		};
-	};
-};
-
-&csi1_bridge {
-    fsl,mipi-mode;
-    status = "okay";
-
-    port {
-        csi1_ep: endpoint {
-            remote-endpoint = <&csi1_mipi_ep>;
-        };
-    };
+	status = "disabled";
 };
 
 &snvs {
@@ -855,17 +752,6 @@
 	status = "okay";
 };
 
-&mipi_dsi {
-	status = "okay";
-	panel: panel@0 {
-		compatible = "startek,kd050hdfia020";
-		reg = <0>;
-		reset-gpio = <&pca9555 6 GPIO_ACTIVE_LOW>;
-		dsi-lanes = <4>;
-		status = "okay";
-	};
-};
-
 &cpu_alert0 {
 	temperature = <105000>;
 };
diff --git a/arch/arm64/boot/dts/compulab/sb-mcm-imx8-rev1.dtsi b/arch/arm64/boot/dts/compulab/sb-mcm-imx8-rev1.dtsi
index e306b1f3adb2..3548e99fded2 100644
--- a/arch/arm64/boot/dts/compulab/sb-mcm-imx8-rev1.dtsi
+++ b/arch/arm64/boot/dts/compulab/sb-mcm-imx8-rev1.dtsi
@@ -14,6 +14,39 @@
 
 #include <dt-bindings/pinctrl/pins-imx8mm.h>
 
+/ {
+	simple_sound: sound {
+		compatible = "simple-audio-card";
+		simple-audio-card,name = "mcm-imx8m-mini";
+		simple-audio-card,widgets =
+			"Headphone", "Headphone Jack",
+			"Line", "Line Out",
+			"Microphone", "Mic Jack",
+			"Line", "Line In";
+		simple-audio-card,routing =
+			"Headphone Jack", "RHPOUT",
+			"Headphone Jack", "LHPOUT",
+			"MICIN", "Mic Bias",
+			"Mic Bias", "Mic Jack";
+		simple-audio-card,format = "i2s";
+		simple-audio-card,bitclock-master = <&sound_master>;
+		simple-audio-card,frame-master = <&sound_master>;
+		simple-audio-card,bitclock-inversion;
+
+		sound_master: simple-audio-card,cpu {
+			sound-dai = <&sai2>;
+			system-clock-frequency = <0>;
+			system-clock-direction = "out";
+		};
+
+		sound_codec: simple-audio-card,codec {
+			sound-dai = <&wm8731>;
+			system-clock-direction = "in";
+			system-clock-type = "mclk";
+		};
+	};
+};
+
 &pcie0 {
 	reset-gpio = <&pca9555 0 GPIO_ACTIVE_LOW>;
 	ext_osc = <1>;
@@ -39,3 +72,118 @@
 		pagesize = <16>;
 	};
 };
+
+&i2c4 {
+	clock-frequency = <400000>;
+	pinctrl-names = "default";
+	pinctrl-0 = <&pinctrl_i2c4>;
+	status = "okay";
+
+	wm8731: wm8731@1a {
+		#sound-dai-cells = <0>;
+		compatible = "wlf,wm8731";
+		reg = <0x1a>;
+		status = "okay";
+	};
+
+	ov5640_mipi: ov5640_mipi@3c {
+		compatible = "ovti,ov5640_mipi";
+		reg = <0x3c>;
+		status = "okay";
+		pinctrl-names = "default";
+		pinctrl-0 = <&pinctrl_csi1>;
+		clocks = <&clk IMX8MM_CLK_CLKO1_DIV>;
+		clock-names = "csi_mclk";
+		assigned-clocks = <&clk IMX8MM_CLK_CLKO1_SRC>,
+				  <&clk IMX8MM_CLK_CLKO1_DIV>;
+		assigned-clock-parents = <&clk IMX8MM_CLK_24M>;
+		assigned-clock-rates = <0>, <24000000>;
+		csi_id = <0>;
+		/*
+		To be looked into & fixed
+		As of now the device is alwas out of reset & under the power
+		*/
+		pwn-gpios = <&pca9555 8 GPIO_ACTIVE_LOW>;
+		rst-gpios = <&pca9555 3 GPIO_ACTIVE_HIGH>;
+
+		mclk = <24000000>;
+		mclk_source = <0>;
+		port {
+			ov5640_mipi1_ep: endpoint {
+				remote-endpoint = <&mipi1_sensor_ep>;
+			};
+		};
+	};
+	goodix_ts@5d {
+		compatible = "goodix,gt911";
+		reg = <0x5d>;
+
+		interrupt-parent = <&gpio1>;
+		interrupts = <1 IRQ_TYPE_LEVEL_HIGH>;
+
+		pinctrl-names = "default";
+		pinctrl-0 = <&touchscreen_pins>;
+
+		irq-gpios = <&gpio1 1 GPIO_ACTIVE_HIGH>;
+		reset-gpios = <&pca9555 4 GPIO_ACTIVE_HIGH>;
+		status = "okay";
+	};
+
+};
+
+&mipi_csi_1 {
+	#address-cells = <1>;
+	#size-cells = <0>;
+	status = "okay";
+	port {
+		mipi1_sensor_ep: endpoint1 {
+			remote-endpoint = <&ov5640_mipi1_ep>;
+			data-lanes = <2>;
+			csis-hs-settle = <13>;
+			csis-clk-settle = <2>;
+			csis-wclk;
+		};
+
+		csi1_mipi_ep: endpoint2 {
+			remote-endpoint = <&csi1_ep>;
+		};
+	};
+};
+
+&csi1_bridge {
+    fsl,mipi-mode;
+    status = "okay";
+
+    port {
+        csi1_ep: endpoint {
+            remote-endpoint = <&csi1_mipi_ep>;
+        };
+    };
+};
+
+&mipi_dsi {
+	status = "okay";
+	panel: panel@0 {
+		compatible = "startek,kd050hdfia020";
+		reg = <0>;
+		reset-gpio = <&pca9555 6 GPIO_ACTIVE_LOW>;
+		dsi-lanes = <4>;
+		status = "okay";
+	};
+};
+
+&flexspi {
+	compatible = "fsl,imx8mm-flexspi";
+	pinctrl-names = "default";
+	pinctrl-0 = <&pinctrl_flexspi0>;
+	status = "okay";
+
+	qspiflash0: qspiflash@0 {
+		reg = <0>;
+		#address-cells = <1>;
+		#size-cells = <1>;
+		compatible = "w25x16", "jedec,spi-nor";
+		spi-max-frequency = <80000000>;
+		spi-nor,ddr-quad-read-dummy = <6>;
+	};
+};
diff --git a/arch/arm64/boot/dts/compulab/sbc-mcm-imx8m-mini.dts b/arch/arm64/boot/dts/compulab/sbc-mcm-imx8m-mini.dts
new file mode 100644
index 000000000000..d08abebf5f2c
--- /dev/null
+++ b/arch/arm64/boot/dts/compulab/sbc-mcm-imx8m-mini.dts
@@ -0,0 +1,9 @@
+/dts-v1/;
+
+#include "mcm-imx8m-mini.dts"
+#include "sb-mcm-imx8-rev1.dtsi"
+
+/ {
+	model = "CompuLab MCM i.MX8MM on SB-MCM8M Rev 1.0";
+	compatible = "cpl,sbc-mcm-imx8m-mini", "cpl,mcm-imx8m-mini", "fsl,imx8mm";
+};
-- 
2.11.0

